common/static/common/js/discussion/views/discussion_inline_view.js

```diff
@@ -212,8 +213,18 @@
         },
 
         hideNewPost: function() {
-            this.$('section.discussion').find('.inline-discussion-thread-container').show();
+            this.$('section.discussion').find('.inline-discussion-thread-container').removeClass('is-hidden');
+            this.$('section.discussion').find('.new-post-btn')
+                .removeClass('is-hidden')
+                .focus();
             return this.newPostForm.addClass('is-hidden');
```

Comment:
Nit: this isn't your change, but the ``return`` is unnecessary and confusing (it is probably from the CoffeeScript conversion).